src/components/FilterDropdown/FilterDropdown.tsx

```diff
@@ -0,0 +1,49 @@
+import ArrowDownIcon from '#/icons/icon-arrowDown-solid.svg';
+import { ReserveSortOption, PriceSortOption } from '@/constants/dropdown';
+import { useState } from 'react';
+import styles from './FilterDropdown.module.css';
+
+interface PriceFilter {
+  type: '가격';
+  lists: PriceSortOption[];
+}
+
+interface ReserveFilter {
+  type: '예약 상태';
+  lists: ReserveSortOption[];
+}
+
+type FilterDropDownProps = PriceFilter | ReserveFilter;
+type AllPriceOption = PriceSortOption | PriceFilter['type'];
+type AllReserveOption = ReserveSortOption | ReserveFilter['type'];
+
+function FilterDropDown({ type, lists }: FilterDropDownProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [value, setValue] = useState<AllPriceOption | AllReserveOption>(type);
+
```

Comment:
```suggestion

interface FilterDropDownProps {
  type: '가격' | '예약 상태';
}

type AllPriceOption = PriceSortOption | '가격';
type AllReserveOption = ReserveSortOption | '예약 상태';

function FilterDropDown({ type }: FilterDropDownProps) {
  const [isOpen, setIsOpen] = useState(false);
  const [value, setValue] = useState<AllPriceOption | AllReserveOption>(type);

  const handleDropdown = () => {
    setIsOpen((prev) => !prev);
  };

  // 내부적으로 type에 따라서 리스트 결정
  const lists = type === '가격' ? PRICE_LIST : RESERVE_LIST;

```

이렇게 하면 불필요한 import와, 타입 정의, prop을 줄일 수 있습니다!